numpy/core/tests/test_indexing.py

```diff
@@ -673,11 +673,7 @@ def __setitem__(self, indx, val):
             k[0:7]
             raise AssertionError
         except ValueError:
-            # The exception holds a reference to the slice so clear on Py2
-            if hasattr(sys, 'exc_clear'):
-                with suppress_warnings() as sup:
-                    sup.filter(DeprecationWarning)
-                    sys.exc_clear()
+            pass
```

Comment:
This and the one below should be refactored into a `with assert_raises(ValueError): k[0:7]`

Reply:
Is `assert_raises` preferred over `pytest.raises`? I see a mix of both throughout the test suit.

Reply:
We moved to pytest about two years ago and still have not changed our style guide to prefer pure pytest syntax over the numpy.testing syntax. Both are fine.